vaultkeeper/tests/test_vaultkeeper.py

```diff
@@ -170,78 +145,46 @@ def test_renew_lease(self):
 
     @responses.activate
     def test_run_natural_success(self, tmpdir):
-        self.vaultkeeper.vault_client.token = '00000000-0000-0000-0000-000000000001'
+        self.fake_gatekeeper.add_handlers(responses, self.fake_gatekeeper_url)
+        self.fake_vault.add_handlers(responses, self.fake_vault_url)
+
+        self.vaultkeeper.vault_client.token = (
+            '00000000-0000-0000-0000-000000000001')
         self.vaultkeeper.vault_secret = vault_token()
         outfile = tmpdir.join(self.vaultkeeper.configs.credential_path)
         self.vaultkeeper.configs.credential_path = outfile.strpath
-        responses.add_callback(responses.POST,
-                               self.fake_gatekeeper_url + '/token',
-                               callback=self.fake_gatekeeper.get_token,
-                               content_type='application/json')
-        responses.add_callback(responses.GET,
-                               self.fake_vault_url
-                               + '/v1/database/creds/postgresql_myschema_readonly',
-                               callback=self.fake_vault.get_db_creds,
-                               content_type='application/json')
-        responses.add_callback(responses.GET,
-                               self.fake_vault_url + '/v1/auth/token/lookup-self',
-                               callback=self.fake_vault.lookup_self,
-                               content_type='application/json')
-
-        responses.add_callback(responses.POST,
-                               self.fake_vault_url + '/v1/auth/token/renew',
-                               callback=self.fake_vault.renew_lease,
-                               content_type='application/json')
-        responses.add_callback(responses.GET,
-                               self.fake_vault_url + '/v1/auth/token/lookup-self',
-                               callback=self.fake_vault.lookup_self,
-                               content_type='application/json')
-
-        responses.add_callback(responses.PUT,
-                               self.fake_vault_url + '/v1/sys/leases/renew',
-                               callback=self.fake_vault.renew_lease,
-                               content_type='application/json')
+
+        if not os.path.exists(os.path.dirname(outfile.strpath)):
+            try:
+                os.makedirs(os.path.dirname(outfile.strpath))
+            except OSError as exc:
+                if exc.errno != errno.EEXIST:
+                    raise
```

Comment:
We know what the config is and we know we're working in a temp dir, so we can create `tmpdir.join('creds')` unconditionally instead of checking for it and catching exceptions. Alternatively, we could configure vaultkeeper with `./creds.txt` instead of a subdir and not worry about creating directories at all.